macOS/DuckDuckGo/WebExtensions/WebExtensionLoader.swift

```diff
@@ -18,39 +18,91 @@
 
 #if WEB_EXTENSIONS_ENABLED
 
+import CryptoKit
+
 @available(macOS 15.4, *)
 protocol WebExtensionLoading: AnyObject {
-
-    func loadWebExtensions(from paths: [String]) async -> [WKWebExtension]
-
+    @discardableResult
+    func loadWebExtension(path: String, into controller: WKWebExtensionController) async throws -> WKWebExtensionContext
+    func loadWebExtensions(from paths: [String], into controller: WKWebExtensionController) async -> [Result<WKWebExtensionContext, Error>]
+    func unloadExtension(at path: String, from controller: WKWebExtensionController) throws
```

Comment:
Several changes here... the loader now takes care of the entirety of loading and unloading onto and from the controller.  This allowed me to also support unloading extensions and simplify the code.